nucypher/policy/conditions/evm.py

```diff
@@ -154,15 +155,17 @@ def _next_endpoint(
         self, providers: Dict[int, Set[HTTPProvider]]
     ) -> Iterator[HTTPProvider]:
         """Yields the next web3 provider to try for a given chain ID"""
-        try:
-            rpc_providers = providers[self.chain]
-
-        # if there are no entries for the chain ID, there
-        # is no connection to that chain available.
-        except KeyError:
-            raise NoConnectionToChain(chain=self.chain)
+        rpc_providers = providers.get(self.chain, None)
         if not rpc_providers:
-            raise NoConnectionToChain(chain=self.chain)  # TODO: unreachable?
+            # only use when no connection available
+            public_rpc_providers = fetch_public_rpc_endpoints_for_chain(
```

Comment:
i think we should call this on node startup, rather than every time a condition is evaluated

Reply:
👍  - That's reasonable. It may play in with whether we want to address the following, which I added in the PR description.
```
Does this functionality also allow for replacement of the data used for default RPC endpoints done in https://github.com/nucypher/nucypher/pull/3496 for officially supported chains.
```

Currently `fetch_public_rpc_endponts_for_chain` (`_fetch_public_rpc_endpoints_from_chainlist`) uses `memoize` so only one HTTP request call is ever made and the result is cached - so the request is not actually made every condition evaluation but rather only the first one. 

Also, we could think about possibly using a time-based cache instead, if we expect the list is refreshed every so often (eg. one per hour). If not, then caching is one time, and to your point, can happen once at startup.

Updated the PR description to have this discussion point.

Reply:
ah nice, love that it's using `memoize` that's perfect

Reply:
Completely agree, I think caching is much better than just a simple call on startup